src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -1,30 +1,49 @@
 import * as React from 'react';
+import { observer } from 'mobx-react';
 import { ButtonGroup, DropdownButton, MenuItem, Toolbar } from 'patternfly-react';
 import { ButtonToolbar } from 'react-bootstrap';
 
 import { GraphFilterProps, GraphFilterState } from '../../types/GraphFilter';
-import * as API from '../../services/Api';
+import Namespace from '../../types/Namespace';
 import { DurationButtonGroup } from './DurationButtonGroup';
 import { LayoutButtonGroup } from './LayoutButtonGroup';
+import { namespaceStore } from '../../model/Namespace';
 
+type DropdownListType = {
+  list: Namespace[];
+  current: Namespace;
+  onSelect: Function;
+};
+
+export class DropdownList extends React.PureComponent<DropdownListType, {}> {
+  render() {
+    return this.props.list.map(ns => (
+      <MenuItem
+        key={ns.name}
+        active={ns.name === this.props.current.name}
+        eventKey={ns.name}
+        onSelect={this.props.onSelect}
+      >
+        {ns.name}
+      </MenuItem>
+    ));
+  }
+}
+
+@observer
 export class GraphFilter extends React.Component<GraphFilterProps, GraphFilterState> {
   constructor(props: GraphFilterProps) {
     super(props);
-    this.state = {
-      availableNamespaces: []
-    };
   }
 
   componentDidMount() {
-    // TODO: [KIALI-436] API.GetNamespaces() is also called in Services component.
-    // We should consolidate them into one.
-    API.GetNamespaces()
-      .then(this.setNamespaces)
-      .catch(error => {
-        this.props.onError(error);
-      });
+    this.reloadData();
```

Comment:
If different components implement this, isn't that going to cause an api call for each component (that uses the namespace)?